tests/lib/rules/singleline-html-element-content-newline.js

```diff
@@ -0,0 +1,439 @@
+        <div>
+          <div>
+          </div>
+        </div>
+      </template>`,
+      options: [{ strict: true }]
+    },
+    {
+      code: `
+      <template>
+        <div>
+          <!-- comment -->
+        </div>
+      </template>`,
+      options: [{ strict: true }]
+    },
+    // self closing
+    `
+      <template>
+        <self-closing />
+      </template>`,
+    // ignores
+    `
+      <template>
+        <pre>content</pre>
+        <pre attr>content</pre>
+        <pre><span attr>content</span></pre>
+        <textarea>content</textarea>
+        <textarea attr>content</textarea>
+        <textarea><span attr>content</span></textarea>
+      </template>`,
+    {
+      code: `
+        <template>
+          <ignore-tag>content</ignore-tag>
+          <ignore-tag attr>content</ignore-tag>
+          <ignore-tag><span attr>content</span></ignore-tag>
+        </template>`,
+      options: [{
+        ignoreNames: ['ignore-tag']
+      }]
+    },
+    // not target
+    `
+      <template>
+        <div>content
+        </div>
+      </template>`,
+    `
+      <template>
+        <div>
+          content</div>
+      </template>`,
+    // Ignore if no closing brackets
+    `
+      <template>
+        <div
+          id=
+          ""
+    `
+  ],
+  invalid: [
+    {
+      code: `
+        <template>
+          <div class="panel">content</div>
+        </template>
+      `,
+      output: `
+        <template>
+          <div class="panel">
+content
+</div>
+        </template>
+      `,
+      errors: [
+        {
+          message: 'Expected 1 line break after closing bracket of the "div" element, but no line breaks found.',
+          line: 3,
+          column: 30,
+          nodeType: 'HTMLTagClose',
+          endLine: 3,
+          endColumn: 30
+        },
+        {
+          message: 'Expected 1 line break before opening bracket of the "div" element, but no line breaks found.',
+          line: 3,
+          column: 37,
+          nodeType: 'HTMLEndTagOpen',
+          endLine: 3,
+          endColumn: 37
+        }
+      ]
+    },
+    {
+      code: `
+        <template>
+          <tr attr><td>singleline</td><td>children</td></tr>
+        </template>
+      `,
+      output: `
+        <template>
+          <tr attr>
+<td>singleline</td><td>children</td>
+</tr>
+        </template>
+      `,
+      errors: [
+        'Expected 1 line break after closing bracket of the "tr" element, but no line breaks found.',
+        'Expected 1 line break before opening bracket of the "tr" element, but no line breaks found.'
+      ]
+    },
+    {
+      code: `
+        <template>
+          <div attr><!-- singleline comment --></div>
+        </template>
+      `,
+      output: `
+        <template>
+          <div attr>
+<!-- singleline comment -->
+</div>
+        </template>
+      `,
+      errors: [
+        'Expected 1 line break after closing bracket of the "div" element, but no line breaks found.',
```

Comment:
Same question as in the other rule, what do you think about using `opening tag` and `closing tag` phrases?